File: readability/readability.c
#include <cs50.h>
#include <string.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>


int count_words(string texto);
int count_letters(string texto);
int sentence(string texto);

int main(void)
{
    int num, nota;
    string texto = get_string ("Text: ");
    int palavras = count_words(texto);
    num = count_letters(texto);
    int sentenca = sentence(texto);
    float grade = 0.0588 * (100 * num / palavras) - 0.296 * (100 * sentenca / palavras) - 15.8;
    float dif = grade - round(grade);
    if(grade>7.30 && grade<7.60){
        nota = floor(grade);
    }
    else{
        int first = round(grade);
        nota = (int)first;
    }

    if(nota>16){
        printf("Grade 16+\n");
    }
    if(nota<1){
        printf("Before Grade 1\n");
    }
    if (nota>1 && nota< 16){
        printf("Grade %d\n", nota);
    }
}

int count_words(string texto){
    int count= 1;
    for(int i=0; i<strlen(texto);i++){
        if(texto[i] == ' '){
            count++;
        }
    }
    return count;
}

int count_letters(string texto){
    int count=0;
    for(int i=0;i<strlen(texto);i++){
        if((texto[i] >= 'a' && texto[i]<='z')||(texto[i] >= 'A' && texto[i]<= 'Z')){
            count++;
        }
    }
    return count;
}

int sentence(string texto){
    int count = 0;
    for(int i = 0; i<strlen(texto); i++){
        if(texto[i] == '.'||texto[i]=='!'||texto[i]=='?'){
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string
from math import *

def count_letters(txt):
    letras = 0
    for i in range(len(txt)):
        if txt[i] == '.' or txt[i] == ' ' or txt[i]==',' or txt[i]=="'" or txt[i] == ':' or txt[i]=='?' or txt[i]=='!':
            continue
        else:
            letras+=1
    return letras

def count_words(txt):
    words = 1
    for i in range(len(txt)):
        if txt[i] == ' ':
            words+=1
        else:
            continue
    return words

def sentences_count(txt):
    sentences = 0
    for i in range(len(txt)):
        if txt[i] == '.' or txt[i] == '!' or txt[i] == '?':
            sentences+=1
        else:
            continue
    return sentences

def main():
    texto = get_string('Texto: ')

    letters = count_letters(texto)
    words = count_words(texto)
    sentences = sentences_count(texto)

    L = float((letters/words)*100)
    S = float((sentences/words)*100)
    grade = (0.0588 * L - 0.296 * S - 15.8)
    grade = round(grade)
    if(grade>16):
        print("Grade 16+\n")

    if(grade<1):
        print("Before Grade 1\n")

    if (grade>1 and grade< 16):
        print(f"Grade {grade}")


main()